docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+```
+
+@:@
+
+**After:**
+
+```
+/My files
+├── /dont delete
+│  
+```
+
+### Exercise
+
+You are tired of people abusing the FTP server to upload long pirated movies. So you decide to implement a method that checks if a file exceeds a certain size, and if so, automatically deletes it (hint: check out the `size` method!). Return `true` if the file was deleted, `false` otherwise:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala
+extension (path: Path) def deleteIfChubby(threshold: Long): IO[Boolean] = ???
+```
+
+@:choice(static)
+
+```scala
+def deleteIfChubby(path: Path, threshold: Long): IO[Boolean] = ???
+```
+
+@:@
+
+[See possible implementation](https://gist.github.com/a4f2ef825f8b3425f41a6069b13a08c0.git)
+
+## Using temporary files
+
+Maybe you do not want to manually delete a file after its use. This is where temporary files come in to play, as they are deleted automatically.
+
+To create temporary files, you have two options, one is to make Cats Effect automatically handle their lifecycle with the `withTempFile` and `withTempDirectory` methods (useful when you want the files deleted right away), or, if you rather prefer the operating system to take hands in its lifecycle, you can use the `createTempFile` and `createTempDirectory` variants (suitable if you do not care if the files are deleted immediately).
+
+The former takes as a parameter a function that describes how you want to use the file, like this:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import cats.syntax.all.*
+import cats.effect.{IO, IOApp}
+
+object Temporary extends IOApp.Simple:
+
+  def run: IO[Unit] = withTempFile: path =>
+    for
+      _ <- path.writeLines(LazyList.from('a').map(_.toChar.toString).take(26))
+      alphabet <- path.read
+      _ <- IO.println("ASCII took hispanics into account!").whenA(alphabet.contains('ñ'))
+      // Not gonna be printed, sadly
+    yield ()
+
+end Temporary
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import cats.syntax.all.*
+import cats.effect.{IO, IOApp}
+
+object Temporary extends IOApp.Simple:
+
+  def run: IO[Unit] = FilesOs.withTempFile: path =>
+    for
+      _ <- FilesOs.writeLines(path, LazyList.from('a').map(_.toChar.toString).take(26))
+      alphabet <- FilesOs.read(path)
+      _ <- IO.println("ASCII took hispanics into account!").whenA(alphabet.contains('ñ'))
+      // Not gonna be printed, sadly
+    yield ()
+
+end Temporary
+```
+
+@:@
+
+You will see that the `use` function goes from `Path => IO[A]`, and that `use` basically describes a path that will be used to compute an `A`, with some side effects along the way. When the computation is finished, the file will no longer exist.
+
+The last alternative is with `createTempFile` or `createTempDirectory`. The difference to the one without `create` is that it returns the path where the file was created, something like this:
```

Comment:
```suggestion
The last alternative is with `createTempFile` or `createTempDirectory`. The difference between `createTempFile` and `withTempFile` is that the `create` functions return the path of the file, for example:
```

Try to be explicit/accurate when explaining differences between terms, it can get easily confusing.